### PR TITLE
refactor: prover handling helpers, remove duplicate code

### PR DESCRIPTION
**Pull request by sergeytimoshin**

### CODE REVIEW

**Comment by SwenSchaeferjohann** [comment]
does this have downstream breaking changes for photon @sergeytimoshin?

File: cli/src/utils/processPhotonIndexer.ts [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+import { killProcess, spawnBinary, waitForServers } from "./process";
+import { INDEXER_PROCESS_NAME, PHOTON_VERSION } from "./constants";
+import { exec } from "node:child_process";
+import * as util from "node:util";
 
-export async function startIndexer() {
-  console.log("Kill existing indexer process...");
-  await killProcessByName(INDEXER_PROCESS_NAME);
+//   const binDir = path.join(__dirname, "../..", "bin");
+//   command = path.join(binDir, binaryName);
```
Comment (SwenSchaeferjohann) [inline cli/src/utils/processPhotonIndexer.ts:8]: are these leftovers ?

Comment (sergeytimoshin) [reply cli/src/utils/processPhotonIndexer.ts]: Yes, thanks, removed.

**Comment by sergeytimoshin** [comment]
> does this have downstream breaking changes for photon @sergeytimoshin?

No, it's just a cleanup.

**Review by SwenSchaeferjohann — APPROVED** [review]

**Pull request merged**